agent/stats.go

```diff
@@ -54,8 +55,18 @@ func (s *statsReporter) callback(_, _ time.Time, virtual, _ map[netlogtype.Conne
 	s.L.Lock()
 	defer s.L.Unlock()
 	s.logger.Debug(context.Background(), "got stats callback")
-	s.networkStats = &virtual
-	s.unreported = true
+	// Accumulate stats until they've been reported.
+	if s.unreported {
+		if s.networkStats == nil && virtual != nil {
+			s.networkStats = make(map[netlogtype.Connection]netlogtype.Counts)
+		}
+		for k, v := range virtual {
+			s.networkStats[k] = s.networkStats[k].Add(v)
+		}
+	} else {
+		s.networkStats = maps.Clone(virtual)
+		s.unreported = true
+	}
```

Comment:
Review: If the callback was called multiple times before reporting, we lost data as each update is a snapshot since the last.

This can happen if:
1. The interval is short (tests)
2. Report takes a long time

Reply:
Great catch!